app/eventyay/webapp/src/components/profile/GreetingPrompt.vue

```diff
@@ -104,14 +104,30 @@ export default {
 	},
 	async created() {
 		this.activeStep = this.steps[0]
-		this.profile = Object.assign({
+		// Determine default display name:
+		// 1. Use existing saved display_name if available (even if empty string)
+		// 2. Otherwise, use wikimedia_username if available
+		// 3. Otherwise, leave empty
+		let defaultDisplayName = ''
+		if (this.user.profile?.display_name != null) {
+			defaultDisplayName = this.user.profile.display_name
+		} else if (this.user.wikimedia_username) {
+			defaultDisplayName = this.user.wikimedia_username
+		}
+
+		// Build profile object, preserving computed display_name
+		this.profile = {
 			greeted: true,
-			display_name: '',
-			avatar: {
+			display_name: defaultDisplayName,
+			avatar: this.user.profile?.avatar || {
 				identicon: this.user.id
 			},
-			fields: {}
-		}, this.user.profile)
+			fields: this.user.profile?.fields || {},
+			...this.user.profile
+		}
+		// Ensure display_name is not overwritten by spread
+		this.profile.display_name = defaultDisplayName
```

Comment:
If you don't want the `display_name` to be overwritten, just change the position of the spread item:

```js
> const d = {a: 1}

> // At tail
> {a: 2, ...d}
{ a: 1 }
> // At head
> {...d, a: 2}
{ a: 2 }
```